Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -35,18 +37,20 @@ class HttpClient {
             let request = try getRequest()
             let requestData = getRequestData(events: events)
 
-            sessionTask = session.uploadTask(with: request, from: requestData) { data, response, error in
-                if error != nil {
-                    completion(.failure(error!))
-                } else if let httpResponse = response as? HTTPURLResponse {
-                    switch httpResponse.statusCode {
-                    case 1..<300:
-                        completion(.success(httpResponse.statusCode))
-                    default:
-                        completion(.failure(Exception.httpError(code: httpResponse.statusCode, data: data)))
+            sessionTask = session.uploadTask(with: request, from: requestData) { [callbackQueue] data, response, error in
```

Comment:
It looks like we are using the same dispatch queue as the main amplitude queue here? If for any reason calls to Amplitude are timing out (eg no network or bad network), all operations to store data won't get a chance to execute right? Possibility of missing data?

Reply:
Ah this only puts it into the main queue after we receive a response from Amplitude is it?

Reply:
Yes - the queue serves as a synchronization point for basically wherever we touch the underlying data files, so any response handling, timed flushes, etc should all happen on the queue (at least until we have a more granular strategy for locking).

This won't block the queue for the duration of the request, it will only run on queue to send a request and handle its response, so we shouldn't be blocked on a bad request. However, because we queue operations instead of immediately executing them, this does introduce a chance that queued operations will be dropped vs executed, though this is the behavior of the previous iOS SDK and the Kotlin SDK.